geniza/annotations/views.py

```diff
@@ -33,6 +41,52 @@ def __init__(self, *args, **kwargs):
         super().__init__(content_type=self.content_type, *args, **kwargs)
 
 
+def parse_annotation_data(request):
```

Comment:
I kept this separate from the actual "setting model attributes" code because of this block in the update annotation view:

https://github.com/Princeton-CDH/geniza/blob/490f1e303c0e0c2d58ecaeff95e50d66c52c77e2/geniza/annotations/views.py#L257-L278

But I imagine this entire block could also happen in the model? It's slightly different behavior from the create annotation view:

https://github.com/Princeton-CDH/geniza/blob/490f1e303c0e0c2d58ecaeff95e50d66c52c77e2/geniza/annotations/views.py#L162-L172

Reply:
Are the main differences are that we always log creation but not always updates? and that we sometimes reindex document? (What are the scenarios where document indexing wouldn't automatically be triggered? It might only be needed if content or label have changed.)

I could see it being more semantic if we had a method on the object that indicates whether there are any meaningful changes (not sure what to call it to differentiate from `has_changed`) that merit logging. Maybe also if there are changes that require reindexing. But not sure how much of your time it's worth shifting things around, could just make some notes about possible future refinements.